pkg/irel/root.go

```diff
@@ -16,12 +16,71 @@
 
 package irel
 
-import "github.com/spf13/cobra"
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
 
-var Root = &cobra.Command{
-	Use:               "irel",
-	Short:             "irel is a tool for relocating container images",
-	Run:               func(cmd *cobra.Command, _ []string) { cmd.Usage() },
-	DisableAutoGenTag: true,
+	"github.com/pivotal/image-relocation/pkg/registry"
+	"github.com/pivotal/image-relocation/pkg/registry/ggcr"
+	"github.com/spf13/cobra"
+)
+
+var (
+	caCertPaths   []string
+	skipTLSVerify bool
+
+	Root = &cobra.Command{
+		Use:               "irel",
+		Short:             "irel is a tool for relocating container images",
+		Run:               func(cmd *cobra.Command, _ []string) { cmd.Usage() },
+		DisableAutoGenTag: true,
+	}
+)
+
+func init() {
+	Root.PersistentFlags().StringSliceVarP(&caCertPaths, "ca-cert-path", "", nil, "Path to CA certificate used to verify TLS certificates for registries (can be specified more than once)")
+	Root.PersistentFlags().BoolVarP(&skipTLSVerify, "skip-tls-verify", "", false, "Skip TLS certificate verification for registries")
 }
 
+func mustGetRegistryClient() registry.Client {
```

Comment:
nit: I would call this method `NewRegistryClient` and pass certPath and skipTLSVerify as flag OR name the method `NewRegistryClientWithSelfSignedCerts`

Reply:
Thanks for your suggestions. A couple of thoughts...

### must-Prefix
Prefixing a function with `must` is a commonly used convention in Go to signal to the caller that the function panics in case of an error. There are plenty of examples in the standard library (e.g. [regexp.MustCompile](https://golang.org/pkg/regexp/#MustCompile)).

### Function Visibility
Your alternative naming suggestions seem to ask to make this function public. What's the reason for that? Nobody outside of package `irel` should really call this function.

### Implicit Use of Globals
I thought about using explicit params instead of implicitly referring to the global variables inside this function. In general, I would have liked to avoid using globals altogether as they are considered suboptimal (potential for side-effects, etc). However, it seems that using globals is somewhat dictated by the way `cobra` handles flags. Now, if we accept that we have to live with globals for these flags, we can at least try to not use those variables in more places than really necessary. Switching to explicit params for the function in question would mean those globals are now referred to in 4 different places, across 4 different files, none of which is the same file that originally declared and initialized the variables. That doesn't sound great to me. 

Sorry for the long-winded reply but I thought it might be helpful to better understand my reasoning. Unless you're feeling really strong about any of this, I would prefer to leave the naming and use of variables as is. But I also don't want this to block the merge of this PR, so if you want, I can change it. Just let me know.

Reply:
@sbawaska I'm happy with the function name, but thanks for asking and obtaining Stev's rationale.